### PR TITLE
Upgrade

### PR DESCRIPTION
**Pull request by ghost**
- upgrade angular, ngrx, other dependencies
- replace tslint with eslint
- remove unnecessary dependencies and script
- update ts and karma configs
- upgrade github workflow
- minor SEO improvements

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,78 +12,76 @@
...
+    "karma": "6.4.1",
+    "karma-chrome-launcher": "3.1.1",
+    "karma-coverage": "2.2.0",
+    "karma-jasmine": "5.1.0",
+    "karma-jasmine-html-reporter": "2.0.0",
+    "ts-node": "10.9.1",
+    "typescript": "4.8.4"
   }
-}
+}
```
Comment (kaladay) [inline package.json]: Missing new line at the end.

File: src/index.html [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    function gtag() { dataLayer.push(arguments); }
+    gtag('js', new Date());
+    gtag('config', 'UA-54069646-2');
+  </script>
+</body>
+
 </html>
```
Comment (kaladay) [inline src/index.html]: Missing new line.

File: src/polyfills.ts [hunk 3]
```diff
@@ -18,15 +18,8 @@
...
-import 'classlist.js'; // Run `npm install --save classlist.js`.
-
-/**
- * Web Animations `@angular/platform-browser/animations`
- * Only required if AnimationBuilder is used within the application and using IE/Edge or Safari.
- * Standard animation support in Angular DOES NOT require any polyfills (as of Angular 6.0).
- */
-import 'web-animations-js'; // Run `npm install --save web-animations-js`.
+// Run `npm install --save classlist.js`.
+// Run `npm install --save web-animations-js`.
```
Comment (kaladay) [inline src/polyfills.ts:22]: What are these commented out commands about?

Like "Run ... classlist.js for IE10/IE11 polyfill support."

File: src/polyfills.ts [hunk 4]
```diff
@@ -60,10 +53,7 @@ import 'zone.js'; // Included with Angular CLI.
 /***************************************************************************************************
  * Load `$localize` onto the global scope - used if i18n tags appear in Angular templates.
  */
-import '@angular/localize/init';
-
-
-import 'ie11-custom-properties';
+ import '@angular/localize/init';
```
Comment (kaladay) [inline src/polyfills.ts]: Why the extra leading space?

File: tsconfig.app.json [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
-  },
-  "files": [
-    "src/main.ts",
-    "src/polyfills.ts"
-  ],
-  "include": [
-    "src/**/*.d.ts"
-  ]
-}
+/* To learn more about this file see: https://angular.io/config/tsconfig. */
```
Comment (kaladay) [inline tsconfig.app.json]: Githubs syntax highlighter is making it look like the comment here is not supported.
I don't understand why it is doing that.
Please double check that this is okay.
I see nothing obviously wrong.

File: tsconfig.spec.json [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
-  "files": [
-    "src/test.ts",
-    "src/polyfills.ts"
-  ],
-  "include": [
-    "src/**/*.spec.ts",
-    "src/**/*.d.ts"
-  ]
-}
+/* To learn more about this file see: https://angular.io/config/tsconfig. */
```
Comment (kaladay) [inline tsconfig.spec.json]: Again, please double check.
Could just be something screwy with Github's syntax highlighter.

**Review by kaladay — CHANGES_REQUESTED** [review]

File: src/polyfills.ts (hunk 3 above)
Comment (ghost) [reply src/polyfills.ts:22]: They were added by ng update via a schematic.

File: src/polyfills.ts (hunk 4 above)
Comment (ghost) [reply src/polyfills.ts]: Not sure. Will fix. Thanks.

File: tsconfig.app.json (hunk 5 above)
Comment (ghost) [reply tsconfig.app.json]: Oh, JSON shouldn't have comments. They are on example JSON for angular projects. I will remove. Thanks.

File: src/polyfills.ts (hunk 3 above)
Comment (kaladay) [reply src/polyfills.ts:22]: Very well, they just seem random.
Its automated so I will choose to overlook.

**Review by kaladay — APPROVED** [review]

**Pull request merged**